### PR TITLE
feat: split package and refactor to nx and pnpm

### PR DESCRIPTION
**Pull request by lorentzfb**
This a big PR that aims to simplify dependencies for the end user.

- Split package @equinor/rdf-graph into 4 packages: @equinor/rdf-graph, @equinor/rdf-graph-go, @equinor/rdf-graph-cy, @equinor/rdf-graph-fg3d
- Convert repo to using pnpm instead of npm
- Introduced nx as the build system
- Introduced Lerna as versioning and publishing manager (replacing release please)

[AB#102727](https://dev.azure.com/EquinorASA/bb9bd8cb-74f7-4ffa-b0cb-60eff0a0be58/_workitems/edit/102727)

### CODE REVIEW

**Review by toftalisk — APPROVED** [review]
A lot of comments, but then again, it was a big PR. I think most of them where positive. Non of the comments are show stoppers, so feel free to fix them later or argue why they don't need fixing.

The work is great 👍 Splitting rdf-graph into new package makes the whole code base easier to reason about and use. It is also great not having to include cy-stuff in go-apps and vice versa.

I also like the work you have done pnpm, it is super fast to get started with playground and test running. I haven't played much with nx, but I am sure it plays a role in build time as well.

Awesome 🎉

File: docs/developers.md [hunk 1]
```diff
@@ -6,18 +6,23 @@
 
 ```sh
 # Make sure you are on version 8 of the npm CLI
-$ node -v && npm -v
+$ node -v && pnpm -v
 v18.12.1
-8.19.2
+7.30.0
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply docs/developers.md:11]: Har oppdatert til siste versjon av pnpm som er 8.1.0.

File: libs/fg3d/src/applyPatch.ts [hunk 2]
```diff
@@ -33,6 +33,10 @@ function addNode(f3d: ForceGraph3DInstance, _node: GraphNodePatch) {
 	f3d.graphData(data);
 }
 
-function addEdge(_f3d: ForceGraph3DInstance, _edge: GraphEdgePatch) {}
+function addEdge(_f3d: ForceGraph3DInstance, _edge: GraphEdgePatch) {
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply libs/fg3d/src/applyPatch.ts:36]: Just fixed an eslint error.

File: libs/cy/src/applyPatch.ts [hunk 3]
```diff
@@ -96,8 +96,7 @@ const addProperty = (cy: Cytoscape.Core, { id, prop }: GraphPropertyPatch) => {
 				break;
 			case 'group':
 				currentElement.move({ parent: value });
-				const parent = cy.getElementById(value);
-				parent.data(nodeTypeKey, NodeType.Parent);
+				cy.getElementById(value).data(nodeTypeKey, NodeType.Parent);
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply libs/cy/src/applyPatch.ts:99]: This was just a eslint error fix.

File: libs/cy/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply libs/cy/.eslintrc.json:1]: In a monorepo we often dont have react in a package, but in another. Then we dont use react specific linting rules globally.

File: libs/core/package.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	},
+	"files": [
+		"dist",
+		"CHANGELOG.md",
+		"README.md",
+		"LICENSE"
+	],
+	"dependencies": {
+		"n3": "^1.16.3"
+	},
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply libs/core/package.json:53]: This is nice!

File: .github/workflows/bump-and-release.yml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    branches:
+      - main
+
+permissions: write-all
+
+jobs:
+  release:
+    name: Publish packages
+    # Do not create release for merged dependabot or Snyk PRs
+    if: ${{ !contains(fromJson('["dependabot[bot]", "snyk-bot"]'), github.event.head_commit.author.name) && github.event.pull_request.merged == true }}
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply .github/workflows/bump-and-release.yml:15]: Agree.

File: apps/playground/index.html [hunk 7]
```diff
@@ -3,7 +3,7 @@
 	<head>
 		<meta charset="UTF-8" />
 		<!-- <link rel="icon" type="image/svg+xml" href="/vite.svg" /> -->
-		<link rel="stylesheet" href="https://eds-static.equinor.com/font/equinor-font.css" />
+		<link rel="stylesheet" href="https://cdn.eds.equinor.com/font/equinor-font.css" />
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply apps/playground/index.html:6]: They have changed the url for eds stuff.

**Comment by lorentzfb** [comment]
@toftalisk Thank your for the thorough review, I really appreciate your effort PR reviews! Pushed some new commits to address your comments.

File: apps/playground/package.json [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+		"3d-force-graph": "^1.70.13",
+		"@equinor/eds-icons": "^0.19.1",
+		"@equinor/rdf-graph": "workspace:*",
+		"@equinor/rdf-graph-cy": "workspace:*",
+		"@equinor/rdf-graph-fg3d": "workspace:*",
+		"@equinor/rdf-graph-go": "workspace:*",
+		"cytoscape": "^3.23.0",
+		"cytoscape-cola": "^2.5.1",
+		"cytoscape-cose-bilkent": "^4.1.0",
+		"cytoscape-dagre": "^2.4.0",
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply apps/playground/package.json:21]: Yes :)

File: README.md [hunk 9]
```diff
@@ -1,41 +1,56 @@
-# @equinor/rdf-graph
+# rdf-graph
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply README.md:1]: Some more updates to the readme has been pushed to this branch!

File: apps/playground/.eslintrc.json [hunk 10]
```diff
@@ -0,0 +1,14 @@
+{
+	"extends": ["../../.eslintrc.json"],
```
[the remark this answers is not in the document]
Comment (lorentzfb) [reply apps/playground/.eslintrc.json:2]: This is a pattern that was recommended by NX. It allows for overriding rules per project if needed. I'm going to introduced NX i raphael as well. This linting pattern is perhaps more relevant in that repo.

**Pull request merged**